src/control/jobs.c

```diff
@@ -86,13 +87,14 @@ dt_job_state_t dt_control_job_get_state(_dt_job_t *job)
   return state;
 }
 
-void dt_control_job_set_params(_dt_job_t *job, void *params)
+void dt_control_job_set_params(_dt_job_t *job, void *params, dt_job_destroy_callback callback)
 {
   if(!job || dt_control_job_get_state(job) != DT_JOB_STATE_INITIALIZED) return;
   job->params = params;
+  job->params_destroy = callback;
 }
 
-void *dt_control_job_get_params(const _dt_job_t *job)
+const void *dt_control_job_get_params(const _dt_job_t *job)
```

Comment:
it was mostly a temporary solution to find all `free(params);` because it would fail to compile with a warning.
I have no preference here, if you say so i will drop const.